File: rotateTest.py
arr = [
    [0, 0, 1, 0],
    [0, 0, 1, 0],
    [0, 1, 1, 0],
    [0, 0, 0, 0],
]


def turnRight(arr):
    lenX = len(arr[0])
    lenY = len(arr)

    arr2 = []

    for i in range(lenX):
        arr2.append([0 for i in range(lenY)])

    for i in range(lenY):
        for j in range(lenX):
            # left -> arr2[lenX-1-j][i] = arr[i][j]
            arr2[j][lenY-1-i] = arr[i][j]
    return arr2



def display(arr):
    for i in arr:
        for j in i:
            print(end=f"{j} ")
        print()


display(turnLeft(arr))

File: rotateTest.js
var arr = [
    [0, 0, 1, 0],
    [0, 0, 1, 0],
    [0, 1, 1, 0],
    [0, 0, 0, 0]
]

function turnRight(arr) {
    var arr2 = [
        [0, 0, 0, 0],
        [0, 0, 0, 0],
        [0, 0, 0, 0],
        [0, 0, 0, 0]
    ]

    for (var i = 0; i < arr.length; i++) {
        for (var j = 0; j < arr[i].length; j++) {
            // Python -> arr2[j][lenY-1-i] = arr[i][j]
            arr2[j][arr.length - 1 - i] = arr[i][j]
        }
    }

    return arr2
}

function turnLeft(arr) {
    var arr2 = [
        [0, 0, 0, 0],
        [0, 0, 0, 0],
        [0, 0, 0, 0],
        [0, 0, 0, 0]
    ]

    for (var i = 0; i < arr.length; i++) {
        for (var j = 0; j < arr[i].length; j++) {
            arr2[arr[i].length-1-j][i] = arr[i][j]
        }
    }

    return arr2
}

console.log(turnLeft(arr))
/* result
[
    [0, 0, 0, 0],
    [0, 1, 0, 0],
    [0, 1, 1, 1],
    [0, 0, 0, 0]
]
*/